src/app/core/models/asset/asset-property.model.ts

```diff
@@ -29,19 +36,19 @@ export abstract class AssetProperty {
     return this.assetId;
   }
 
-  isAddress(): boolean {
+  isAddress(): this is Address {
```

Comment:
What is this sorcery? Please change it back to `boolean` so we have proper type hinting.